otherlibs/systhreads/st_stubs.c

```diff
@@ -545,6 +546,18 @@ CAMLprim value caml_thread_initialize(value unit)
     caml_failwith("caml_thread_initialize: cannot initialize Thread "
                   "while several domains are running.");
 
+  thread_table = caml_stat_alloc_noexc(caml_params->max_domains *
+                                       sizeof(struct caml_thread_table));
+  if (thread_table == NULL)
+    caml_fatal_error("caml_thread_initialize: failed to allocate thread"
+                     " table");
+
+  tick_thread_stop = caml_stat_alloc_noexc(caml_params->max_domains *
+                                           sizeof(atomic_uintnat));
+  if (tick_thread_stop == NULL)
+    caml_fatal_error("caml_thread_initialize: failed to allocate tick"
+                     " thread stop");
```

Comment:
Could we store `tick_thread_stop` in `thread_table` instead? This is not strictly-speaking related to the current PR, but it would simplify it noticeably and may improve other things in the future.

(In OCaml 4 we had a single global variable. The choice to make it a separate domain-indexed array was made very early in the threads-on-multicore implementation work, 7994082605daf9d39de55c57d5450e74c221838d ).